client/errors.go

```diff
@@ -11,15 +11,16 @@ import (
 
 // errConnectionFailed implements an error returned when connection failed.
 type errConnectionFailed struct {
-	host string
+	host  string
+	cause error
 }
 
 // Error returns a string representation of an errConnectionFailed
 func (err errConnectionFailed) Error() string {
 	if err.host == "" {
-		return "Cannot connect to the Docker daemon. Is the docker daemon running on this host?"
+		return "Cannot connect to the Docker daemon. Is the docker daemon running on this host?: " + err.cause.Error()
```

Comment:
Perhaps we need to `errors.Wrap()` the error or implement the `Causer` interface (but to be discussed)